best-practices.mdx

```diff
@@ -0,0 +1,190 @@
+---
+title: Best Practices
+description: Learn about our best practices for building and maintaining your Chalk solution
+---
+
+# Best Practices
+
+With Chalk, the same solution can be implemented in a number of different ways. Below are some guidelines
+illustrating recommended patterns for building and maintaining your Chalk solution!
+
+## Data Sources/Integrations
+
+### Test your datasource connections
+
+Define integrations through the Chalk dashboard and check that they're connecting properly using the test datasource button.
+![Test Datasource](/img/test-datasource.png)
+
+### Avoid naming your datasources by their "type", e.g. don't call your postgres datasource "postgres"
+
+Though this works, it can lead to ambiguity in SQL resolvers when you have multiple data sources of the same type. This occurs because only one instance of a data source type, [Chalk lets you refer to that data source by it's type instead of its name](/fraud-3#online-data)
+
+## Features
+
+### Start by defining your features
+
+Features fully specify what you want your data to look like. Once you have an understanding of the inner relations, writing resolvers for your features becomes easier.
+
+### Avoid dataclass feature types, instead use seperate features sets or unpack the data
+
+While Chalk allows for dataclass feature types, they should be avoided. They don't always play nicely with serialization and can cause tough to debug errors. We recommend
+either unpacking the nested class into basic types or defining and joining an additionally Chalk feature set if the nested component is truly a separate entity. Defining a
+separate feature set, also makes the underlying data easier to monitor and test.
+
+### Tag and annotate your features
+
+We strongly recommend that you tag and annotate your features as you are developing them. This is done by adding comments above a feature, like so:
```

Comment:
Annotation might be more important than tagging? Not sure if all basic customers will have ideas of how to tag features

Reply:
Now that I am wriitng this out, we should also explain why this is useful and what we can do with tags and annotations.